appintro/src/main/java/com/github/paolorotolo/appintro/AppIntroBase.java

```diff
@@ -451,11 +482,48 @@ private void handleSlideChanged(Fragment oldFragment, Fragment newFragment) {
 
     /**
      * Called after a new slide has been selected
+     * {@link #canChangeLastSlide(int)} is called to check if the last slide can be changed.
      *
      * @param position Position of the new selected slide
      */
     protected void onPageSelected(int position) {
-        // Empty method
+        LogHelper.d(TAG, "Selected slide number: " + (position + 1));
+        canChangeLastSlide(position);
+    }
+
+
+    /**
+     * Called by {@link #onPageSelected(int)} to check if the user has requested permissions on the previous slide.
+     * This method is required as a safety check because otherwise,
```

Comment:
@cortinico, look the problem is there is a weird bug without this check. Everything is fine but if I swipe a little bit backward and 
then swipe forward fast, the slide gets changed without requesting permissions, so that's why I put this check in place. Have a look in this video: